main.go

```diff
@@ -0,0 +1,192 @@
+	recipes := []interface{}{eloteRecipe, locoMocoRecipe, patatasBravasRecipe, friedRiceRecipe}
+	insertManyResult, err := collection.InsertMany(context.TODO(), recipes)
+	if err != nil {
+		fmt.Println("Something went wrong trying to insert the new documents:")
+		panic(err)
+	}
+	fmt.Println("the following documents were inserted:", insertManyResult.InsertedIDs)
+
+	/*
+	 * *** FIND DOCUMENTS ***
+	 *
+	 * Now that we have data in Atlas, we can read it. To retrieve all of
+	 * the data in a collection, we create a filter for recipes that take
+	 * less than 45 minutes to prepare and sort by name (ascending)
+	 */
+
+	var filter = bson.M{"prepTimeInMinutes": bson.M{"$lt": 45}}
+	options := options.Find()
+
+	// Sort by `name` field ascending
+	options.SetSort(bson.D{{"name", 1}})
+
+	cursor, err := collection.Find(context.TODO(), filter, options)
+	if err != nil {
+		fmt.Println("Something went wrong trying to find the documents:")
+		panic(err)
+	}
+
+	defer func() {
+		cursor.Close(context.Background())
+	}()
+
+	// Loop through the returned recipes
+	for cursor.Next(ctx) {
+		recipe := Recipe{}
+		err := cursor.Decode(&recipe)
+
+		// If there is an error decoding the cursor into a Recipe
+		if err != nil {
+			fmt.Println("cursor.Next() error:")
+			panic(err)
+		} else {
+			fmt.Println(recipe.Name, "has", len(recipe.Ingredients), "ingredients, and takes", recipe.PrepTimeInMinutes, "minutes to make.")
+		}
+	}
+
+	// We can also find a single document. Let's find the first document
+	// that has the string "fried rice" in the name.
+	var result Recipe
+	var myFilter = bson.D{{"name", "fried rice"}}
+	e := collection.FindOne(context.TODO(), myFilter).Decode(&result)
+	if e != nil {
+		fmt.Println("Something went wrong trying to find one document:")
+		panic(e)
+	}
+	fmt.Println("found a document with the name fried rice", result)
+
+	/*
+	 * *** UPDATE A DOCUMENT ***
+	 *
+	 * You can update a single document or multiple documents in a single call.
+	 *
+	 * Here we update the prepTimeInMinutes value on the document we
+	 * just found.
+	 */
+
+	var updateDoc = bson.D{{"$set", bson.D{{"prepTimeInMinutes", 72}}}}
+	var myRes = collection.FindOneAndUpdate(ctx, myFilter, updateDoc, nil)
+	if myRes.Err() != nil {
+		fmt.Println("Something went wrong trying to update one document:")
+		panic(myRes.Err())
+	}
+	doc := bson.M{}
+	decodeErr := myRes.Decode(&doc)
+	if decodeErr != nil {
+		fmt.Println("Something went wrong trying to decode the document:")
+		panic(decodeErr)
+	}
+	fmt.Println("Here is the updated document:", doc)
+
+	/*      *** DELETE DOCUMENTS ***
+	 *
+	 *      As with other CRUD methods, you can delete a single document
+	 *      or all documents that match a specified filter. To delete all
+	 *      of the documents in a collection, pass an empty filter to
+	 *      the DeleteMany() method. In this example, we'll delete two of
+	 *      the recipes.
+	 */
+
+	deletedRecipeNameList := make([]string, 0)
```

Comment:
Is there some need to use a verbose form of initializing an array? I might see some value if there were comments on each line to point out something specific, but I think the multiline comment above about this section explains what's going on here. You can replace ll. 181-183 with:

```
deletedRecipeNameList := [...]string{"elotes", "fried rice"}
```

Reply:
👍